config/mapprovider.go

```diff
@@ -110,26 +128,62 @@ func NewRetrievedFromMap(cfgMap *Map, opts ...RetrievedOption) Retrieved {
 	for _, opt := range opts {
 		opt(&set)
 	}
-	return Retrieved{cfgMap: cfgMap, closeFunc: set.closeFunc}
+	return &valueRetrieved{value: cfgMap.ToStringMap(), CloseFunc: set.closeFunc}
 }
 
-// AsMap returns the retrieved configuration parsed as a Map.
-func (r Retrieved) AsMap() (*Map, error) {
-	return r.cfgMap, nil
+// NewRetrievedFromStringMap returns a new Retrieved instance that contains a map[string]interface{} value.
+// * cfgMap the Map that will be merged to the given map in the MergeTo.
+// * CloseFunc specifies a function to be invoked when the configuration for which it was
+//   used to retrieve values is no longer in use and should close and release any watchers
+//	 that it may have created.
+func NewRetrievedFromStringMap(value map[string]interface{}, opts ...RetrievedOption) Retrieved {
+	set := retrievedSettings{}
+	for _, opt := range opts {
+		opt(&set)
+	}
+	return &valueRetrieved{value: value, CloseFunc: set.closeFunc}
 }
 
-// Close and release any watchers that MapProvider.Retrieve may have created.
-//
-// Should block until all resources are closed, and guarantee that `onChange` is not
-// going to be called after it returns except when `ctx` is cancelled.
-//
-// Should never be called concurrently with itself.
-func (r Retrieved) Close(ctx context.Context) error {
-	if r.closeFunc == nil {
-		return nil
+// NewRetrievedFromString returns a new Retrieved instance that contains a string value.
+// * cfgMap the Map that will be merged to the given map in the MergeTo.
+// * CloseFunc specifies a function to be invoked when the configuration for which it was
+//   used to retrieve values is no longer in use and should close and release any watchers
+//	 that it may have created.
+func NewRetrievedFromString(value string, opts ...RetrievedOption) Retrieved {
+	set := retrievedSettings{}
+	for _, opt := range opts {
+		opt(&set)
+	}
+	return &valueRetrieved{value: value, CloseFunc: set.closeFunc}
+}
+
+type valueRetrieved struct {
+	value interface{}
+	CloseFunc
+}
+
+func (r *valueRetrieved) AsMap() (*Map, error) {
+	val, ok := r.value.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("retrieved value (type=%v) cannot be used as a *Map", reflect.TypeOf(r.value))
```

Comment:
You can [use `%T`](https://pkg.go.dev/fmt#hdr-Printing) in a format string to get the type
```suggestion
		return nil, fmt.Errorf("retrieved value (type=%T) cannot be used as a *Map", r.value)
```